routers/web/org/home.go

```diff
@@ -48,32 +48,20 @@ func Home(ctx *context.Context) {
 	ctx.Data["Title"] = org.DisplayName()
 
 	var orderBy db.SearchOrderBy
-	ctx.Data["SortType"] = ctx.FormString("sort")
-	switch ctx.FormString("sort") {
-	case "newest":
-		orderBy = db.SearchOrderByNewest
-	case "oldest":
-		orderBy = db.SearchOrderByOldest
-	case "recentupdate":
-		orderBy = db.SearchOrderByRecentUpdated
-	case "leastupdate":
-		orderBy = db.SearchOrderByLeastUpdated
-	case "reversealphabetically":
-		orderBy = db.SearchOrderByAlphabeticallyReverse
-	case "alphabetically":
-		orderBy = db.SearchOrderByAlphabetically
-	case "moststars":
-		orderBy = db.SearchOrderByStarsReverse
-	case "feweststars":
-		orderBy = db.SearchOrderByStars
-	case "mostforks":
-		orderBy = db.SearchOrderByForksReverse
-	case "fewestforks":
-		orderBy = db.SearchOrderByForks
-	default:
-		ctx.Data["SortType"] = "recentupdate"
+
+	sortOrder := strings.ToLower(ctx.FormString("sort"))
+	if sortOrder == "" {
+		// TODO: add new default sort order for org home?
+		sortOrder = setting.UI.ExploreDefaultSort
+	}
+
+	if order, ok := repo_model.OrderByFlatMap[sortOrder]; ok {
+		orderBy = order
+	} else {
+		sortOrder = "recentupdate"
 		orderBy = db.SearchOrderByRecentUpdated
 	}
+	ctx.Data["SortType"] = sortOrder
```

Comment:
I think it could be as simple as:

```suggestion
	sortOrder := ctx.FormString("sort")
	if _, ok := repo_model.OrderByFlatMap[sortOrder]; !ok {
		sortOrder = setting.UI.ExploreDefaultSort // TODO: add new default sort order for org home?
	}
	ctx.Data["SortType"] = sortOrder
	orderBy = repo_model.OrderByFlatMap[sortOrder]
```

And everything should work well.

Reply:
Is it ok to remove `strings.ToLower`?

Reply:
Why it needs `strings.ToLower`?

Reply:
It comes from https://github.com/go-gitea/gitea/pull/30876/files#diff-2223ddaf610a594e44153435be6477904e56dbfca54317b35e7ada231a9ab7faR61

Reply:
How could it be right?

Reply:
Maybe user will input the URL manually? Then we can adjust the value which maybe wrong.
I think there are no big differences, so I also added it here.

@6543 
Why did you add `strings.ToLower` in #30876. 🤔

Reply:
because it's  also used in the api witch where we do not control the sort paramerter, so to make the api option more resiliant it is used.

here as we should control the var via our own frontend, it is not nessesary, through you could argue input sanitazation never hurts (if not to resource intensive...)

Reply:
@yp05327 https://github.com/go-gitea/gitea/pull/31951 well looking twice i noticed that it was on the wrong place ...